packages/core/src/api/toDecimal.ts

```diff
@@ -49,12 +49,17 @@ function getDecimal<TAmount>(
   formatter: Formatter<TAmount>
 ) {
   const absoluteFn = absolute(calculator);
+  const equalFn = equal(calculator);
+  const lessThanFn = lessThan(calculator);
+  const zero = calculator.zero();
 
   return (units: readonly TAmount[], scale: TAmount) => {
-    return units
+    const lastIndex = units.length - 1;
+
+    const decimalString = units
       .map((unit, index) => {
         const isFirst = index === 0;
-        const isLast = units.length - 1 === index;
+        const isLast = lastIndex === index;
```

Comment:
I think we could also drop `isFirst` and `isLast` in favor of comparing directly to `index` in the following lines. `index === 0` and `index === lastIndex` are pretty self-descriptive. I don't know if using constants is more v8-friendly, though.

Reply:
I'm fine with inlining as long as it doesn't hurt readability, as the performance gains are usually very small.

If you think that works well here, go for it.